docs/content/preview/develop/build-apps/java/ysql-vertx-pg-client.md

```diff
@@ -0,0 +1,214 @@
+                .setPassword("yugabyte");
+
+            Vertx vertx = Vertx.vertx();
+            // creating the PgPool with configuration as option and maxsize 10.
+            PgPool pool = PgPool.pool(vertx, options, new PoolOptions().setMaxSize(10));
+
+            //getting a connection from the pool and running the example on that
+            pool.getConnection().compose(connection -> {
+                Promise<Void> promise = Promise.promise();
+                // create a test table
+                connection.query("create table test(id int primary key, name text)").execute()
+                        .compose(v -> {
+                            // insert some test data
+                            return connection.query("insert into test values (1, 'Hello'), (2, 'World'), (3,'Example'), (4, 'Vertx'), (5, 'Yugabyte')").execute();
+                        })
+                        .compose(v -> {
+                            // prepare the query
+                            return connection.prepare("select * from test order by id");
+                        })
+                        .map(preparedStatement -> {
+                            // create a stream for the prepared statement
+                            return preparedStatement.createStream(50, Tuple.tuple());
+                        })
+                        .onComplete(ar -> {
+                            if (ar.succeeded()) {
+                                RowStream<Row> stream = ar.result();
+                                stream
+                                        .exceptionHandler(promise::fail)
+                                        .endHandler(promise::complete)
+                                        .handler(row -> System.out.println(row.toJson())); // Printing each row as JSON
+                            } else {
+                                promise.fail(ar.cause());
+                            }
+                        });
+                return promise.future().onComplete(v -> {
+                    // close the connection
+                    connection.close();
+                });
+            }).onComplete(ar -> {
+                if (ar.succeeded()) {
+                    System.out.println("Example ran successfully!");
+                } else {
+                    ar.cause().printStackTrace();
+                }
+            });
+
+        }
+    }
+    ```
+
+1. Run your new program.
+
+    ```sh
+    $ mvn -q package exec:java -DskipTests -Dexec.mainClass=com.yugabyte.vertxPgExample
+    ```
+
+    You should see the following as the output:
+
+    ```output
+    {"id":1,"name":"Hello"}
+    {"id":2,"name":"World"}
+    {"id":3,"name":"Example"}
+    {"id":4,"name":"Vertx"}
+    {"id":5,"name":"Yugabyte"}
+    Example ran successfully!
+    ```
+## Limitation
+
+[PubSub](https://vertx.io/docs/vertx-pg-client/java/#_pubsub) feature of Vertx PG client is currently not supported with YugabyteDB.
```

Comment:
If there's a GitHub issue associated with this limitation, include it.

Reply:
@priyanshi-yb You can include this issue: #1872